datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -855,24 +850,18 @@ fn build_recover_project_plan(
     Projection::try_new(col_exprs, Arc::new(input)).map(LogicalPlan::Projection)
 }
 
-fn extract_expressions(
-    expr: &Expr,
-    schema: &DFSchema,
-    result: &mut Vec<Expr>,
-) -> Result<()> {
+fn extract_expressions(expr: &Expr, result: &mut Vec<Expr>) {
```

Comment:
I may have missed it, but all the usages of extract_expressions seem to be for a new empty vec

So this could potentially be simplified even further (perhaps as a follow on PR)

```rust
fn extract_expressions(expr: &Expr) -> Vec<Expr> {
```

Reply:
The call inside the for loop below might pass a non-empty vector.
https://github.com/apache/datafusion/blob/21e35eeb4e106835cc059205315d94ad0201a921/datafusion/optimizer/src/common_subexpr_eliminate.rs#L537-L544